hazelcast/src/main/java/com/hazelcast/multimap/impl/operations/RemoveBackupOperation.java

```diff
@@ -50,29 +49,20 @@ public void run() throws Exception {
             return;
         }
         Collection<MultiMapRecord> coll = multiMapValue.getCollection(false);
-        Iterator<MultiMapRecord> iterator = coll.iterator();
-        while (iterator.hasNext()) {
-            if (iterator.next().getRecordId() == recordId) {
-                iterator.remove();
-                response = true;
-                if (coll.isEmpty()) {
-                    container.delete(dataKey);
-                }
-                break;
-            }
-        }
+        MultiMapRecord record = new MultiMapRecord(isBinary() ? value : toObject(value));
+        response = coll.remove(record);
```

Comment:
You should probably leave this part in:
```java
                if (coll.isEmpty()) {
                    container.delete(dataKey);
                }
```

Reply:
@mmedenjak, correct, I’ll fix it in a both operations.